src/compositor/lipstickcompositor.cpp

```diff
@@ -732,3 +737,30 @@ void LipstickCompositor::readContent()
 {
     m_recorder->recordFrame(this);
 }
+
+void LipstickCompositor::surfaceCommitted()
+{
+    if (isVisible() || m_fakeRepaintTriggered)
+        return;
+
+    static const int gracePeriod = qEnvironmentVariableIsSet("LIPSTICK_GRACE_PERIOD_DURATION") ? qgetenv("LIPSTICK_GRACE_PERIOD_DURATION").toInt() : 2000;
+    if (gracePeriod >= 0 && (!m_graceTimer.isValid() || m_graceTimer.elapsed() > gracePeriod)) {
+        m_graceTimer.invalidate();
+        return;
+    } else if (gracePeriod < 0) {
+        m_graceTimer.invalidate();
+    }
+
+    startTimer(100);
+    m_fakeRepaintTriggered = true;
+}
+
+void LipstickCompositor::timerEvent(QTimerEvent *e)
+{
+    Q_UNUSED(e)
+
+    frameStarted();
+    sendFrameCallbacks(surfaces());
```

Comment:
Is it a problem if display comes back on and timer is still active, causing one fake repaint?